cmd/login.go

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/netflix/weep/pkg/util"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func init() {
+	loginCmd.PersistentFlags().BoolVarP(&noOpen, "no-open", "x", false, "print the link, but do not open a browser window")
+	rootCmd.AddCommand(loginCmd)
+}
+
+var loginCmd = &cobra.Command{
+	Use:   "login",
+	Short: loginShortHelp,
+	Long:  loginLongHelp,
+	Args:  cobra.MaximumNArgs(1),
+	RunE:  runLogin,
+}
+
+func runLogin(cmd *cobra.Command, args []string) error {
+	// If a role was provided, use it, otherwise prompt
+	role, err := InteractiveRolePrompt(args, region, nil)
+	if err != nil {
+		return err
+	}
+
+	// Construct the URL and open/print it; default to HTTPS if not specified
+	url := fmt.Sprintf("%s/role/%s", viper.GetString("consoleme_open_url_override"), role)
```

Comment:
This should fall back to `consoleme_url` if `consoleme_open_url_override` isn't set. I think the cleanest way to do this would be to refactor this function into the config package and export it: https://github.com/Netflix/weep/blob/5ef5b4a1df4ae350925c7a8f11913fbf445937ef/pkg/creds/consoleme.go#L213-L220

Reply:
Makes sense, I will give that a try.